Bottom: 4cc67c45a1f87dcea7bf98526bed8a70433c7ee6
Top:    8ee616766db7f977548d4e719d03d17506b1cef7
Author: Ian Kent <raven@themaw.net>
Date:   2013-06-07 15:00:57 +0800

autofs-5.0.7 - fix incorrect check in flag_is_owned()

The flag file code isn't used any more but this is clearly incorrect
so fix it in case it gets used sometime in the future.


---

diff --git a/daemon/flag.c b/daemon/flag.c
index f8fe163..db9a4bd 100644
--- a/daemon/flag.c
+++ b/daemon/flag.c
@@ -66,12 +66,11 @@ static int flag_is_owned(int fd)
 
 			continue;
 		}
-
-		/* Stale flagfile */
-		if (!tries)
-			return 0;
 	}
 
+	/* Stale flagfile */
+	if (!tries)
+		return 0;
 
 	if (pid) {
 		int ret;
